libs/platform/experience/src/plugins/preview-plugin.ts

```diff
@@ -0,0 +1,28 @@
+import { AppPlugin, AppPluginBeforeApply } from '@spryker-oryx/core';
+import { rootInjectable } from '@spryker-oryx/utilities';
+import { isServer } from 'lit';
+
+export const PreviewPluginName = 'oryx.experience$preview';
+
+export class PreviewPlugin implements AppPlugin, AppPluginBeforeApply {
+  getName(): string {
+    return PreviewPluginName;
+  }
+
+  apply(): void {
+    //
+  }
+
+  beforeApply(): void {
+    const root = rootInjectable.get();
+
+    if (!isServer && root === 'body') {
+      return;
+    }
+
+    if (!isServer) {
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```

Comment:
shouldn't we check, if there is some pre-rendered content before rewritting element?